drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb

```diff
@@ -600,105 +550,180 @@ def self.question_number
+      process_scope_by_client question_name, table_1_exits, SHE_COLUMNS do |_client, client_enrollments|
         # 2. Of the project exits selected in step 1, determine the latest
         # project exit for each client.
-        last_exit = client_enrollments.max_by { |e| e[:date] }
-
-        # 3. If the latest exit was from a PH project (types 3, 9 and 10) where
-        # the [housing move-in date] is <= [report end date], exclude the client
-        # completely from this measure (the client will be reported in measure
-        # 7b.2). If the latest exit was from project types 3, 9 and 10 and there
-        # is no [housing move-in date] for that stay, the client is included in
-        # this measure.
-        next if last_exit[:project_type].in?(PH_PSH) && last_exit[:MoveInDate].present? && last_exit[:MoveInDate].to_date <= @report.end_date
+        last_exit = client_enrollments.max_by { |e| e[:last_date_in_program] }
+        #  puts "#{table_1_dest_field} checking #{last_exit}"
+
+        # 3. If the latest exit was from a PH project (types 3, 9 and 10)
+        # where the [housing move-in date] is <= [report end date], exclude the client completely
+        # from this measure (the client will be reported in measure 7b.2).
+        # If .... there is no [housing move-in date] for that stay, the client is included in this measure.
+        if last_exit[:project_type].in?(PH_PSH) && last_exit[:MoveInDate].present? && last_exit[:MoveInDate] <= @report.end_date
+          # puts "EXCLUDED #{last_exit} in Step 3 from #{table_1_dest_field}"
+          next
+        end
 
         # 4. Reference the destinations of the project exits against Appendix A
         # (row headers) using the project type from which the exit occurred
         # (column headers). Destinations indicated with an X (values 15, 6, 25,
         # 24) cause leavers with those destinations to be completely excluded
         # from the entire measure (all of column C).
-        next if last_exit[:destination].in?(m7b_excluded_destinations)
+        excluded_destinations = if last_exit[:project_type] == 3 # SO
+          [6, 29, 24]
+        else
+          [15, 6, 25, 24]
+        end
+        if last_exit[:project_type].in? excluded_destinations
+          # puts "EXCLUDED #{last_exit} in Step 4 from #{table_1_dest_field}"
+          next
+        end
+        # M6 variant "a without a housing move-in date"
+        if m_code == :m6 && last_exit[:MoveInDate].present?
+          # puts "EXCLUDED #{last_exit} in Step 1 from #{table_1_dest_field}"
+          next
+        end
 
+        # Steps 5 - 7 are handled in the subclass
         {
-          fields[:table_1] => last_exit[:destination] || 99,
-          fields[:history] => client_enrollments,
-        }
+          table_1_dest_field => last_exit[:destination] || 99,
+          history_field => client_enrollments,
+        }.tap do |data|
+          # puts "FOUND #{last_exit} for #{table_1_dest_field}"
```

Comment:
This feels unnecessary, if it's useful for debugging, maybe we should just comment out the entire tab block?

Reply:
Sure. i can remove it entire but i've had to take it out and put it back multiple times to debug so figured it was easiest to leave the hook in a way that `rubocop -a` doesnt need to reformat every time